test/unit-tests/expression/parse.test.js

```diff
@@ -2328,6 +2369,13 @@ describe('parse', function () {
         assert.strictEqual(parseAndEval('(true or true) and false'), false)
       })
 
+      it('should respect precedence between logical nand and or', function () {
```

Comment:
Nice range of tests! This precedence test, though, needs adjusting because the cases you happened to choose don't distinguish the precedence options. For example, both of these expressions are true:  `(false nand true) or true` and `false nand (true or true)`. For testing precedence, you need an expression for which the result depends on how it is parenthesized. For example, for a nand - or combo like the first one, you could use `true nand false or true`, since `(true nand false) or true` is true, but `true nand (false or true)` is false.  You should find operands that show the same contrast for the other combos you test. Overall, those combos should include at least nand - and, nand-or, nor - and, and nor - or.  Thanks!